astropy/coordinates/angles/core.py

```diff
@@ -161,7 +161,7 @@ def __new__(cls, angle, unit=None, dtype=np.inexact, copy=True, **kwargs):
                     "will be interpreted simply as a sequence with the given unit."
                 )
 
-            elif isinstance(angle, str):
+            if isinstance(angle, str):
```

Comment:
What was wrong with the way the code was before?

Reply:
Looks like this linter rule doesn't like an `else` statement after a `raise`. See: https://docs.astral.sh/ruff/rules/superfluous-else-raise/